maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx

```diff
@@ -301,7 +301,7 @@ export const LayerControlPanel = memo(
                               <EuiFlexItem>
                                 <EuiListGroupItem
                                   key={layer.id}
-                                  label={layer.name}
+                                  label={layer.name || layersTypeNameMap[layer.type]}
```

Comment:
For pending new layer, how about name `"New " + layersTypeNameMap[layer.type]`

Reply:
Synced with Vijay, let's have input from UX here

Reply:
Updated new layer name as New layer <number>